ansible_base/resource_registry/tasks/sync.py

```diff
@@ -148,22 +148,105 @@ def get_resource_type_names() -> list[str]:
     return [f"shared.{rt.model._meta.model_name}" for _, rt in sorted(resources.items())]
 
 
+def _handle_conflict(resource_data: dict, resource_type: ResourceType, api_client: ResourceAPIClient):
+    # How might we end up with a conflict?
+    # - get_orphan_resources ignores resources that are owned by the service and have
+    #   is partially migrated set to false. We could be conflicting with one of those.
+    # - You could rename A --> B and B --> C. If A is updated before B is, then we will
+    #   end up needing to update B before we can update A.
+    # - You could rename A --> B and B --> A. If this happens, there is really no recovery.
+    #   Updating either would result in a conflict. We would have to set A or B to some
+    #   temporary value. Hopefully this will never happen since this scenario would be
+    #   incredibly difficult to pull off.
+
+    conflict_resource = resource_type.get_conflicting_resource(resource_data)
+
+    if conflict_resource is None:
+        return
+
+    resp = api_client.get_resource(conflict_resource.ansible_id)
+
+    # If the conflicting resource doesn't exist on the server, just go ahead and delete it.
+    # This will most likely happen with resources that are ignored by get_orphan_resources.
+    if resp.status_code == 404:
+        delete_resource(conflict_resource)
+
+    # If the resource does exist, lets update it first. Hopefully this desn't also result
+    # in a duplicate key error. If it does, we're cooked.
+    elif resp.status_code == 200:
+        data = resp.json()
+        conflict_resource.update_resource(
```

Comment:
Do we want to try/catch here or otherwise validate that we succeeded?

Reply:
If this does raise an integrity error, or any other DB related error it will get caught in `_attempt_create_resource` or `_attempt_update_resource`.